examples/src/main/kotlin/io/ethers/examples/batchrequests/BatchRequests.kt

```diff
@@ -0,0 +1,105 @@
+package io.ethers.examples.batchrequests
+
+import UniswapV2Pair
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.providers.HttpClient
+import io.ethers.providers.Provider
+import io.ethers.providers.types.BatchRpcRequest
+import io.ethers.providers.types.await
+import io.ethers.providers.types.batchRequest
+import io.ethers.providers.types.resultOrThrow
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import java.math.BigDecimal
+import java.math.BigInteger
+
+/**
+ * Request batching of pool reserves on different DEXes for WETH / USDC pair and calculating WETH price,
+ * showcasing both consolidated and manual batching methods.
+ */
+class BatchRequests(
+    httpRpcUrl: String,
+    uniPoolAddr: String,
+    sushiPoolAddr: String,
+) {
+    // Init provider
+    private val httpClient = HttpClient(httpRpcUrl)
+    private val provider = Provider(httpClient)
+
+    // Init pair contracts
+    private val uniPool = UniswapV2Pair(provider, Address(uniPoolAddr))
+    private val sushiPool = UniswapV2Pair(provider, Address(sushiPoolAddr))
+
+    fun run() {
+        var uniWethPrice: BigDecimal
+        var sushiWethPrice: BigDecimal
+
+        // Consolidated request batching
+        var (uniReserves, sushiReserves) = batchRequest(
+            uniPool.getReserves().call(BlockId.LATEST),
+            sushiPool.getReserves().call(BlockId.LATEST),
+        ).await().resultOrThrow()
+
+        uniWethPrice = formatUnits(uniReserves._reserve0, 6) / formatUnits(uniReserves._reserve1, 18)
```

Comment:
Conversion to BigDecimal be simplified with, e.g. `uniReserves._reserve0.toBigDecimal(6)`